src/plopp/functions/inspector.py

```diff
@@ -0,0 +1,85 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2022 Scipp contributors (https://github.com/scipp)
+
+from .common import require_interactive_backend, preprocess
+from .figure import figure
+from ..core import input_node, widget_node, Node
+from ..core.utils import coord_as_bin_edges
+from ..widgets import PointsTool
+
+from scipp import Variable, scalar
+from scipp.typing import VariableLike
+from numpy import ndarray
+from typing import Union, Dict
+
+
+def inspector(obj: Union[VariableLike, ndarray],
+              dim: str = None,
+              *,
+              crop: Dict[str, Dict[str, Variable]] = None,
+              **kwargs):
+    """
+    """
+    require_interactive_backend('inspector')
+
+    from plopp.widgets import SliceWidget, slice_dims, Box
+    da = preprocess(obj, crop=crop, ignore_size=True)
+
+    a = input_node(da)
+
+    if dim is None:
+        dim = da.dims[-1]
+
+    # Convert dimension coords to bin edges
+    for d in (set(da.dims) - set([dim])):
+        da.coords[d] = coord_as_bin_edges(da, d)
+
+    sl = SliceWidget(da, dims=[dim])
+    w = widget_node(sl)
+
+    slice_node = slice_dims(a, w)
+    f2d = figure(slice_node)
+    xdim = f2d._dims['x']['dim']
+    ydim = f2d._dims['y']['dim']
+
+    f1d = figure()
+
+    event_nodes = {}
+
+    def make_node(change):
```

Comment:
What is the reason for the design choice of nested functions?

Reply:
No particular reason. It was the easiest. I can refactor.